articles/matlab-image-database/index.md

```diff
@@ -0,0 +1,120 @@
+
+The `num2str` function provides the filename. This means that since the count is starting from one up to ‘n’ which is the no. of given frames, so filenames will be in the form of numbers.
+In the filename above, we gave the file format to the `png`. This means that we expect the name to be in the form below;
+imageDatabase1.png
+imageDatabase2.png
+imageDatabase3.png until
+imageDatabasen.png  
+
+![this is how the image database appears](/engineering-education/matlab-image-database/database.png)
+
+If there is no face detected in the cropped frame, then it should display `no face detected` in the workspace. Here is the code;
+
+```Matlab
+else
+disp(‘no face detected’)
+```
+The camera makes a snapshot after 0.1s and resets the camera after the snapshot is made. This is regulated by the code below;
+
+```Matlab 
+clf(‘reset’)
+pause(0.1)
+end 
+```
+The pause time can be changed depending on the user's preference.
+After this, the camera object is cleared.
+When you run the program, the face of the person in front of the camera is captured and stored. This is repeated until the number of the required images are obtained. The number of captured faces is displayed in the workspace.
+When the person in front of the camera hides his/her face or when there is nobody in front of the camera, then a text `no face is detected` is displayed on the workspace. When no face is detected, the counting stops, and continuous immediately face is detected until it obtains the required samples.
+
+`myFaceDetect` function should be defined in a different script and then called into the face database script. 
+The function returns a cropped image and bboxpoints but at this point, we major on the bboxpoints.
+
+```Matlab
+function [croppedImage, bboxpoints] = myFaceDetect
+%introduce the face detector object
+
+faceDetector = vision.CascadeObjectDetector;
+faceDetector.MergeThreshold = 10; %adjust to avoid false detection
+%add bounding boxbox around the detect face
+bboxes = faceDetector(img)
+
+if ~isempty(bboxes)
+bboxes(1,1) = bboxes(1,1)-50;
+bboxes(1,2) = bboxes(1,2)-50;
+bboxes(1,3) = bboxes(1,3)+100;
+bboxes(1,4) = bboxes(1,4)+100;
+```
+
+In the code above we add the value and increase the size of the bounding box in both directions. This allows the other parts of the body such as the ear to be included. 
+
+```Matlab
+%crop the obtained image
+croppedImage = imcrop(img,bboxes)
+bboxPoints = bbox2points(bboxes(1, :))
+%if face is not detected the code below is executed
+else
+croppedImage = [];
+bboxPoints = [];
+end
+```
+
+### Conclusion.
+Matlab is used to create an image database. This is because it has a built-in webcam. It also has a feature that allows you to use your phone as an IPCAM. This makes it efficient for creating a database for analysis. The image database can also be used in the deep neural system for training and validation. This can help in object recognition and other projects.
```

Comment:
Add a `Further reading` section, that contains references or links that the reader can learn more about this topic